packages/core/src/amazonqFeatureDev/session/session.ts

```diff
@@ -166,7 +166,7 @@ export class Session {
     }
 
     public async insertChanges() {
-        for (const filePath of this.state.filePaths ?? []) {
+        for (const filePath of this.state.filePaths?.filter(i => !i.rejected) ?? []) {
```

Comment:
Can we add a test for this specific scenario as well? A test will catch any future regressions for something that is important functionality

Reply:
I will add it as a follow up, I want to merge it as soon as possible to mitigate the bug

Reply:
The tests are supposed to provide confidence that the bug was mitigated. Have we done some manual testing to ensure the fix works? I don't see any testing details in the PR description.

Reply:
my bad, updated description. Yes, I manually tested. I will add a test in this PR

Reply:
updated descriptions, added tests